action/protocol/poll/slasher.go

```diff
@@ -465,27 +465,24 @@ func (sh *Slasher) calculateUnproductiveDelegates(ctx context.Context, sr protoc
 			bcCtx.Tip.Height,
 			sh.productivity,
 		)
-		if err != nil {
-			return nil, err
-		}
-		// The current block is not included, so add it
-		numBlks++
-		if _, ok := produce[blkCtx.Producer.String()]; ok {
-			produce[blkCtx.Producer.String()]++
-		} else {
-			produce[blkCtx.Producer.String()] = 1
-		}
 	} else {
 		numBlks, produce, err = rp.ProductivityByEpoch(
 			epochNum,
-			bcCtx.Tip.Height+1,
+			bcCtx.Tip.Height,
 			func(start, end uint64) (map[string]uint64, error) {
 				return currentEpochProductivity(sr, start, end, sh.numOfBlocksByEpoch)
 			},
 		)
-		if err != nil {
-			return nil, err
-		}
+	}
+	if err != nil {
+		return nil, err
+	}
+	// The current block is not included, so add it
+	numBlks++
```

Comment:
so this applies to both if{} and else{}? 
can we also add test case to cover this?